integrationtests/mayavi/common.py

```diff
@@ -383,12 +386,18 @@ def run_standalone(self):
             type, value, tb = self.exception_info
             raise type, value, tb
 
+
     def run(self):
         """This starts everything up and runs the test.  Call main to
         run the test."""
         # Calls the users test code.
         try:
-            self.do()
+            if self.profile:
+                memory_assistant = MemoryAssistant()
+                memory_assistant.assertReturnsMemory(self.do_profile,
```

Comment:
I am discussing the design extension of the integration tests framework. It looks weird that one has to replicate almost the same code once in do and once do_profile. A better api would be that we always perform `self.do()` but we use the memory assistant when the usecase is designated as a memory leak check. 

In short we should keep the `test_contour.py` and we add another called `test_contour_memory.py` (or in the same file but separate test cases).